debian/patches/0069-add-tonemap-videotoolbox.patch

```diff
@@ -1777,6 +1792,34 @@ new file mode 100644
 +    if (ctx->trc_out != AVCOL_TRC_SMPTE2084) {
 +        av_frame_remove_side_data(output, AV_FRAME_DATA_MASTERING_DISPLAY_METADATA);
 +        av_frame_remove_side_data(output, AV_FRAME_DATA_CONTENT_LIGHT_LEVEL);
++    } else {
++        CGColorSpaceRef colorspace = NULL;
++        CFStringRef colormatrix = kCVImageBufferYCbCrMatrix_ITU_R_2020;
++        CFStringRef colorpri = kCVImageBufferColorPrimaries_ITU_R_2020;
++        CFStringRef colortrc = kCVImageBufferTransferFunction_SMPTE_ST_2084_PQ;
++        CFMutableDictionaryRef attachments = CFDictionaryCreateMutable(NULL, 4,
++                                                                       &kCFTypeDictionaryKeyCallBacks,
++                                                                       &kCFTypeDictionaryValueCallBacks);
++        if (!attachments) {
++            err = AVERROR(ENOMEM);
++            goto fail;
++        }
++        CFDictionarySetValue(attachments, kCVImageBufferYCbCrMatrixKey, colormatrix);
++        CFDictionarySetValue(attachments, kCVImageBufferColorPrimariesKey, colorpri);
++        CFDictionarySetValue(attachments, kCVImageBufferTransferFunctionKey, colortrc);
++        colorspace = CVImageBufferCreateColorSpaceFromAttachments(attachments);
++        if (colorspace) {
++            CFDictionarySetValue(attachments, kCVImageBufferCGColorSpaceKey, colorspace);
++            CFRelease(colorspace);
++        } else {
++            av_log(avctx, AV_LOG_WARNING, "Unable to set proper colorspace for the CVImageBuffer.\n");
++        }
++        CVBufferSetAttachments(
++            (CVPixelBufferRef)output->data[3],
++            attachments,
++            kCVAttachmentMode_ShouldPropagate);
++        CFRelease(attachments);
++        ff_update_hdr_metadata(output, 100.0f);
```

Comment:
> ff_update_hdr_metadata(output, 100.0f);

I wasn't sure if this was good enough since dovi meta provides per-scene brightness. Add it for cuda/ocl too?

Reply:
This is reserved for future use because for DOVI to PQ direct passthrough without tonemapping will still cause washed out color as a lot of dovi scene is mastered at 4000nit, if we want to ever implement this, we need to tonemap it to 1000nit HDR10 range, so this value should always be 1000nit here.

Reply:
And in this case, BT2390 would be the only viable method as all other methods lacks a proper way to handle target peak.